build/Dockerfile.autobuild

```diff
@@ -0,0 +1,30 @@
+FROM registry.access.redhat.com/ubi8/go-toolset:1.13.15 AS builder
+USER root
+WORKDIR /scratch
+COPY . .
+
+# Rather than echo the contents of terminal_check_unit.go as in the Mafilefile, we just copy it since the
+# file in sirupsen is essentially identical to what is being generated.  If that changes, so will this.
+RUN go mod vendor &&\
+    cp vendor/github.com/sirupsen/logrus/terminal_check_unix.go vendor/github.com/Sirupsen/logrus/terminal_check_unix.go
+
+RUN go build -o build/_output/bin/kfctl -mod=vendor github.com/kubeflow/kfctl/v3/cmd/manager
```

Comment:
So this is equivalent to `operator-sdk` run?

Reply:
@vpavlin yes, essentially, based on watching operator-sdk verbose output. The SDK command looks like this:

```bash
DEBU[0000] Running []string{"go", "build", "-o", "/home/tmckay-repos/tmckayus-opendatahub-operator/build/_output/bin/tmckayus-opendatahub-operator", "-gcflags", "all=-trimpath=/home/tmckay-repos", "-asmflags", "all=-trimpath=/home/tmckay-repos", "-mod=vendor", "github.com/kubeflow/kfctl/v3/cmd/manager"}
```
I could add the trimpath back in for /scratch -- I originally tried to build this in /, then moved to scratch for perms.
The trimpath would help with debugging if we ever pulled the binary out of the image to debug.

The image build looks like this:
```bash
DEBU[0001] Running []string{"docker", "build", "-f", "build/Dockerfile", "-t", "kubeflow-operator", "--build-arg", "binary_name=tmckayus-opendatahub-operator", "."}
```
The only difference here is the passing of binary_name as a build arg, which we do not need (binary is hardcoded)